tensorboard/plugins/debugger/interactive_debugger_plugin.py

```diff
@@ -96,8 +99,32 @@ def listen(self, grpc_port):
         interactive_debugger_server_lib.InteractiveDebuggerDataServer(
             self._grpc_port))
 
-    threading.Thread(target=self._debugger_data_server.
-                     start_the_debugger_data_receiving_server).start()
+    self._server_thread = threading.Thread(
+        target=self._debugger_data_server.
+        start_the_debugger_data_receiving_server)
+    self._server_thread.start()
+
+    signal.signal(signal.SIGINT, self.signal_handler)
+
+  def signal_handler(self, unused_signal, unused_frame):
```

Comment:
It might be good to note somewhere that this is required because of arguably a deficiency in grpc/python 2.7:
https://github.com/grpc/grpc/issues/6999 (They marked it as wontfix somewhat unreasonably, I think, given that the debugger plugin is exactly that kind of "rare case" where this has an effect in the real world.)  E.g. in python 3, Ctrl+C will successfully kill TensorBoard (actually, it requires two of them in a row, but still it's closer).

Reply:
Added comments. Thanks.